pkg/e2etest/e2e_test.go

```diff
@@ -0,0 +1,529 @@
+	"github.com/els0r/goProbe/pkg/goprobe/writeout"
+	"github.com/els0r/goProbe/pkg/results"
+	"github.com/els0r/goProbe/pkg/types"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/stretchr/testify/require"
+
+	slimcap "github.com/fako1024/slimcap/capture"
+	"github.com/fako1024/slimcap/capture/afpacket/afring"
+	"github.com/fako1024/slimcap/capture/pcap"
+	"github.com/fako1024/slimcap/link"
+)
+
+const testDataPath = "testdata"
+
+//go:embed testdata/*
+var pcaps embed.FS
+
+var defaultCaptureConfig = config.CaptureConfig{
+	Promisc:             false,
+	RingBufferBlockSize: 1048576,
+	RingBufferNumBlocks: 4,
+}
+
+var externalPCAPPath string
+
+func TestE2EBasic(t *testing.T) {
+	pcapData, err := pcaps.ReadFile(filepath.Join(testDataPath, "default.pcap.gz"))
+	require.Nil(t, err)
+
+	testE2E(t, pcapData)
+}
+
+func TestE2EMultipleIfaces(t *testing.T) {
+
+	// Load identical data several times
+	ifaceData := make([][]byte, 3)
+	for i := 0; i < len(ifaceData); i++ {
+		pcapData, err := pcaps.ReadFile(filepath.Join(testDataPath, "default.pcap.gz"))
+		require.Nil(t, err)
+		ifaceData[i] = pcapData
+	}
+
+	testE2E(t, ifaceData...)
+}
+
+func TestE2EExtended(t *testing.T) {
+	pcapDir, err := pcaps.ReadDir(testDataPath)
+	require.Nil(t, err)
+
+	for _, dirent := range pcapDir {
+		path := filepath.Join(testDataPath, dirent.Name())
+
+		t.Run(path, func(t *testing.T) {
+			pcapData, err := pcaps.ReadFile(path)
+			require.Nil(t, err)
+
+			testE2E(t, pcapData)
+		})
+	}
+}
+
+func TestE2EExtendedPermuted(t *testing.T) {
+	pcapDir, err := pcaps.ReadDir(testDataPath)
+	require.Nil(t, err)
+
+	Perm(pcapDir, func(de []fs.DirEntry) {
+		ifaceData := make([][]byte, 0)
+
+		var testMsg string
+		for _, dirent := range de {
+			path := filepath.Join(testDataPath, dirent.Name())
+			testMsg += " " + path
+
+			pcapData, err := pcaps.ReadFile(path)
+			require.Nil(t, err)
+			ifaceData = append(ifaceData, pcapData)
+		}
+
+		t.Run(testMsg, func(t *testing.T) {
+			testE2E(t, ifaceData...)
+		})
+	})
+}
+
+func TestE2EExternal(t *testing.T) {
+	if externalPCAPPath == "" {
+		t.SkipNow()
+	}
+
+	stat, err := os.Stat(externalPCAPPath)
+	require.Nil(t, err)
+	if stat.IsDir() {
+
+		require.Nil(t, filepath.WalkDir(externalPCAPPath, func(path string, d fs.DirEntry, err error) error {
+			if d.IsDir() {
+				return nil
+			}
+			if err != nil {
+				return err
+			}
+
+			fmt.Println("Running E2E test on", path)
+			pcapData, err := os.ReadFile(filepath.Clean(path))
+			require.Nil(t, err)
+
+			testE2E(t, pcapData)
+
+			return nil
+		}))
+
+	} else {
+		fmt.Println("Running E2E test on", externalPCAPPath)
+		pcapData, err := os.ReadFile(filepath.Clean(externalPCAPPath))
+		require.Nil(t, err)
+
+		testE2E(t, pcapData)
+	}
+}
+
+func testE2E(t *testing.T, datasets ...[]byte) {
+
+	// Setup a temporary directory for the test DB
+	tempDir, err := os.MkdirTemp(os.TempDir(), "goprobe_e2e")
+	if err != nil {
+		panic(err)
+	}
+	config.SetRuntimeDBPath(tempDir)
```

Comment:
I'm still at odds with this one (ever since I've introduced it) and hope that we find a better way of doing that in the future. Let's have a look during refactor.